aten/src/ATen/cuda/detail/OffsetCalculator.cuh

```diff
@@ -28,15 +28,11 @@ struct OffsetCalculator {
   // the strides will be in # of elements.
   OffsetCalculator(int dims, const int64_t* sizes, const int64_t* const* strides, const int64_t* element_sizes=nullptr) : dims(dims) {
     TORCH_CHECK(dims <= MAX_DIMS, "tensor has too many (>", MAX_DIMS, ") dims");
-    for (int i = 0; i < MAX_DIMS; ++i) {
-      if (i < dims) {
-        sizes_[i] = IntDivider<index_t>(sizes[i]);
-      } else {
-        sizes_[i] = IntDivider<index_t>(1);
-      }
+    for (int i=0; i < dims; i++){
+      sizes_[i] = IntDivider<index_t>(sizes[i]);
       for (int arg = 0; arg < NARGS; arg++) {
         int64_t element_size = (element_sizes == nullptr ? 1LL : element_sizes[arg]);
-        strides_[i][arg] =  i < dims ? strides[arg][i] / element_size : 0;
+        strides_[i][arg] = strides[arg][i] / element_size;
       }
```

Comment:
btw, might be worth emitting two loops here depending on whether `element_sizes == nullptr` to avoid a bunch of `idiv` instructions just to divide by 1.

Reply:
Tried it, it actually increases instruction count, and in most cases `element_sizes` is not `nullptr`. 
It is silly that TI `build` multiplies by element sizes and stores stride in bytes, and here we copy element_sizes into array (true story, https://github.com/pytorch/pytorch/blob/5610e8271b6cdaef9edc3b609cb806c4d91d8145/aten/src/ATen/native/cuda/Loops.cuh#L43) and then divide by them, but that's outside of this fix scope.